tests/CrossValidation/Metrics/BrierScoreTest.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Rubix\ML\Tests\CrossValidation\Metrics;
+
+use Rubix\ML\Tuple;
+use Rubix\ML\CrossValidation\Metrics\ProbabilisticMetric;
+use Rubix\ML\CrossValidation\Metrics\BrierScore;
+use PHPUnit\Framework\TestCase;
+use Generator;
+
+/**
+ * @group Metrics
+ * @covers \Rubix\ML\CrossValidation\Metrics\BrierScore
+ */
+class BrierScoreTest extends TestCase
+{
+    /**
+     * @var \Rubix\ML\CrossValidation\Metrics\BrierScore
+     */
+    protected $metric;
+
+    /**
+     * @before
+     */
+    protected function setUp() : void
+    {
+        $this->metric = new BrierScore();
+    }
+
+    /**
+     * @test
+     */
+    public function build() : void
+    {
+        $this->assertInstanceOf(ProbabilisticMetric::class, $this->metric);
+        $this->assertInstanceOf(BrierScore::class, $this->metric);
+    }
+
+    /**
+     * @test
+     */
+    public function range() : void
+    {
+        $tuple = $this->metric->range();
+
+        $this->assertInstanceOf(Tuple::class, $tuple);
+        $this->assertCount(2, $tuple);
+        $this->assertGreaterThan($tuple[0], $tuple[1]);
+    }
+
+    /**
+     * @test
+     * @dataProvider scoreProvider
+     *
+     * @param list<array<string,int|float>> $probabilities
+     * @param list<string|int> $labels
+     * @param float $expected
+     */
+    public function score(array $probabilities, array $labels, float $expected) : void
+    {
+        [$min, $max] = $this->metric->range()->list();
+
+        $score = $this->metric->score($probabilities, $labels);
+
+        $this->assertThat(
+            $score,
+            $this->logicalAnd(
+                $this->greaterThanOrEqual($min),
+                $this->lessThanOrEqual($max)
+            )
+        );
+
+        $this->assertEqualsWithDelta($expected, $score, 1e-8);
+    }
+
+    /**
+     * @return \Generator<mixed[]>
+     */
+    public function scoreProvider() : Generator
+    {
+        yield [
+            [
+                ['U' => 0.7, 'O' => 0.3],
+                ['U' => 0.4, 'O' => 0.6],
+                ['U' => 0.2, 'O' => 0.8],
+            ],
+            ['U', 'O', 'U'],
+            -0.5933333333333334,
+        ];
+
+        yield [
+            [
+                ['cat' => 0.6, 'frog' => 0.1, 'hamster' => 0.3],
+                ['cat' => 0.1, 'frog' => 0.7, 'hamster' => 0.2],
+                ['cat' => 0.2, 'frog' => 0.0, 'hamster' => 0.9],
+                ['cat' => 0.5, 'frog' => 0.1, 'hamster' => 0.4],
+                ['cat' => 0.0, 'frog' => 0.8, 'hamster' => 0.2],
+            ],
+            ['cat', 'frog', 'hamster', 'cat', 'frog'],
+            -0.19000000000000003,
+        ];
+
+        yield [
+            [
+                [1 => 0.0, 2 => 0.0, 3 => 1.0],
+                [1 => 0.0, 2 => 1.0, 3 => 0.0],
+                [1 => 1.0, 2 => 0.0, 3 => 0.0],
+            ],
+            [3, 2, 1],
+            0.0,
+        ];
+
+        yield [
+            [
+                [1 => 0.2, 2 => 0.2, 3 => 0.0],
```

Comment:
Are these probabilities correct? (0.4 in total)

> Metric assumes probabilities are between 0 and 1 and their joint distribution sums to 1.

Should we have a specification to validate a sum?

Reply:
That's a good idea but I think it would add too much overhead (we'd have to check every joint probability distribution array). This was a typo on my part.